packages/host/tests/integration/search-index-test.gts

```diff
@@ -3363,6 +3395,21 @@ posts/ignore-me.json
       );
     });
 
+    test('can use an eq filter with a date field', async function (assert) {
```

Comment:
this new test actually breaks in the main branch exposing the existing bug that prevents us from searching against date fields that this PR now fixes